File: hw1/hw1/Calculator.cs
namespace Hw1
{
    internal static class Calculator
    {
        internal static int Calculate(Program.Operation operation, int val1, int val2)
        {
            var result = 0;
            switch (operation)
            {
                case Program.Operation.Add:
                    result = val1 + val2;
                    break;
                case Program.Operation.Subtract:
                    result = val1 - val2;
                    break;
                case Program.Operation.Multiply:
                    result = val1 * val2;
                    break;
                case Program.Operation.Divide:
                    result = val1 / val2;
                    break;
            }
            return result;
        }
    }
}

File: hw5/hw5/Calculator.fs
module hw5.Calculator

open hw5.Operation

let Calculate (val1: decimal, operation:Operation,  val2: decimal) =
    match operation with
    | Operation.Add -> val1 + val2
    | Operation.Subtract -> val1 - val2
    | Operation.Divide -> val1/val2
    | _ -> val1*val2